src/os/bluestore/BlueFS.cc

```diff
@@ -996,6 +1001,39 @@ int BlueFS::_check_new_allocations(const bluefs_fnode_t& fnode,
   return 0;
 }
 
+int BlueFS::_check_delete_allocations(const bluefs_fnode_t& fnode,
```

Comment:
Am I right that _check_new_allocations and _check_delete_allocations differ in the following lines only?
if (!bs.test(pos)) {
  fail = true;
}
bs.reset(pos);
vs.
if (bs.test(pos)) {
  fail = true;
}
bs.set(pos);

If so may be simplify that and leave a single _check_allocations() function getting 'expected' parameter:
if (bs.test(pos) != expected) {
  fail = true;
}
bs.flip(pos);

Reply:
True. Those functions are complementary.
And you are right - melding them together improved readability.